sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseDecodeData.java

```diff
@@ -59,11 +59,24 @@ default Type getHeadersType() {
      * 2. If the returned int[] is not-null, only the codes in the array are considered as success code.
      *
      * @return the expected HTTP response status codes
+     * @deprecated Use {@link #isExpectedResponseStatusCode(int)} instead.
      */
+    @Deprecated // This is only used in tests now - all uses should go via isExpectedResponseStatusCode instead
```

Comment:
This interface is in implementation, so we should be able to remove it completely.

Reply:
Done